proof-systems/src/darlin/benches/accumulate_verify.rs

```diff
@@ -0,0 +1,209 @@
+use algebra::{AffineCurve, ToConstraintField};
+use poly_commit::{
+    PolynomialCommitment,
+    ipa_pc::InnerProductArgPC
+};
+use proof_systems::darlin::{
+    tests::{
+        get_keys,
+        final_darlin::generate_test_data as generate_final_darlin_test_data
+    },
+    proof_aggregator::{accumulate_proofs, verify_aggregated_proofs},
+};
+use digest::Digest;
+use criterion::*;
+use rand::{thread_rng, SeedableRng};
+use blake2::Blake2s;
+use proof_systems::darlin::pcd::GeneralPCD;
+use rand_xorshift::XorShiftRng;
+
+fn bench_verify<G1: AffineCurve, G2: AffineCurve, D: Digest>(
+    c: &mut Criterion,
+    bench_name: &str,
+    segment_size: usize,
+    max_proofs: Vec<usize>,
+)
+    where
+        G1: AffineCurve<BaseField = <G2 as AffineCurve>::ScalarField> + ToConstraintField<<G2 as AffineCurve>::ScalarField>,
+        G2: AffineCurve<BaseField = <G1 as AffineCurve>::ScalarField> + ToConstraintField<<G1 as AffineCurve>::ScalarField>,
+{
+    let rng = &mut XorShiftRng::seed_from_u64(1234567890u64);
+    let mut group = c.benchmark_group(bench_name);
+    let num_constraints = 1 << 19;
+
+    //Generate DLOG keys
+    let params_g1 = InnerProductArgPC::<G1, D>::setup(segment_size - 1).unwrap();
+    let params_g2 = InnerProductArgPC::<G2, D>::setup(segment_size - 1).unwrap();
+
+    let (
+        committer_key_g1, verifier_key_g1,
+        committer_key_g2, verifier_key_g2
+    ) = get_keys::<_, _, D>(&params_g1, &params_g2);
+
+
+    let (final_darlin_pcd, index_vk) = generate_final_darlin_test_data::<G1, G2, D, _>(
+        num_constraints,
+        segment_size,
+        &params_g1,
+        &params_g2,
+        1,
+        rng
+    );
+
+    // Generate proofs and bench
+    for num_proofs in max_proofs.into_iter() {
+
+        // Collect PCDs and vks
+        let pcds = vec![GeneralPCD::FinalDarlin(final_darlin_pcd[0].clone()); num_proofs];
```

Comment:
Does this define a "constant" vector, taking only the first PCD?

Reply:
Ah, ok, we also generated just a single PCD. 
I suppose this is to prevent long preparations?

Reply:
Exactly. I'm interested in testing accumulation/verification so I can as well generate a single proof and clone as many times as I need (otherwise it would really take a loooong time)